amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/MessagePublishContext.java

```diff
@@ -89,16 +89,17 @@ public void recycle() {
      * publish amqp message to pulsar topic, no batch.
      */
     public static void publishMessages(Message<byte[]> message, Topic topic, CompletableFuture<Position> future) {
+        ByteBuf headerAndPayload = messageToByteBuf(message);
         CompletableFuture<PositionImpl> positionFuture = new CompletableFuture<>();
         positionFuture.whenComplete((position, throwable) -> {
             if (throwable != null) {
                 future.completeExceptionally(throwable);
             } else {
                 future.complete(position);
             }
+            headerAndPayload.release();
         });
 
-        ByteBuf headerAndPayload = messageToByteBuf(message);
```

Comment:
We could release the `headerAndPayload` after the method `publishMessage`.

like this
```
ByteBuf headerAndPayload = messageToByteBuf(message);
try {
    topic.publishMessage(
            headerAndPayload, MessagePublishContext.get(positionFuture, topic, System.nanoTime()));
} finally {
    headerAndPayload.release();
}
```

Reply:
ok

Reply:
@gaoran10 Finish.